docs/install/kustomize-structure.md

```diff
@@ -0,0 +1,37 @@
+# About the kustomization structure
+
+The kustomization directories are slightly more complex than the average operator due to the dependancy on `cass-operator`. We deploy `cass-operator` using the manifests from Github to ensure that we remain up to date.
+
+## Objectives
+
+We want to achieve the following:
+
+1. Minimal modifications to the `config/` directory initially created by `operator-sdk`.
+2. All remote operator manifests are pulled from their home GH repositories, not kept locally.
+3. All local operator manifests come directly from `config/`.
+
+## Design
+
+The kustomization directories are as follows. Any kustomizations not generated by `operator-sdk` are stored in `aux-config/` rather than the templated `config/` directory. This is where we deploy k8ssandra-operator from when we want to include all its dependancies. 
```

Comment:
Why is it a different top directory? Why isn't it called ``overlay`` like the tool itself calls it?

Reply:
You're saying you'd rather call `aux-config` `overlay` instead? I'm fine with that, maybe it is better. I don't think it is particularly clear though, but as someone with familiarity with kustomize I guess I might look at it...

I want a different top directory because `config` is scaffolded by the frameworks. I want our customisations to be clearly separated so we don't end up doing bad things like installing `cass-operator` from the `kustomization.yaml` in default, which was what triggered off the errors I saw. 

If you propose to put a new directory `overlays` in `config` I disagree. We lose clarity on whether `default` or `overlays` should be the entrypoint for kustomize builds in this scenario.

Reply:
The last option is what kustomize exactly proposes. 

config/overlays
config/base
config/etc

Reply:
@Miles-Garnsey @burmanm please move the discussion to https://forum.k8ssandra.io/t/establishing-conventions-with-kustomize/338. I think it has gone beyond the original scope of the PR and is better suited for the forum.